docs/developer-docs/backend/rust/12-searching-records.md

```diff
@@ -0,0 +1,390 @@
+    });
+    PROFILE_STORE.with(|profile_store| {
+        profile_store.borrow_mut().insert(principal_id, profile);
+    });
+}
+
+#[query(manual_reply = true)]
+fn search(text: String) -> ManualReply<Option<Profile>> {
+    let text = text.to_lowercase();
+    PROFILE_STORE.with(|profile_store| {
+        for (_, p) in profile_store.borrow().iter() {
+            if p.name.to_lowercase().contains(&text) || p.description.to_lowercase().contains(&text)
+            {
+                return ManualReply::one(Some(p));
+            }
+
+            for x in p.keywords.iter() {
+                if x.to_lowercase() == text {
+                    return ManualReply::one(Some(p));
+                }
+            }
+        }
+        ManualReply::one(None::<Profile>)
+    })
+}
+```
+
+Save your changes and close the file to continue.
+
+## Update the interface description file
+
+Candid is an interface description language (IDL) for interacting with canisters running on the Internet Computer. Candid files provide a language-independent description of a canister’s interfaces including the names, parameters, and result formats and data types for each function a canister defines.
+
+By adding Candid files to your project, you can ensure that data is properly converted from its definition in Rust to run safely on the Internet Computer blockchain.
+
+To see details about the Candid interface description language syntax, see the [Candid Guide](./../candid/index.md) or the [Candid crate documentation](https://docs.rs/candid/).
+
+To update Candid file for this guide open the `src/rust_profile_backend/rust_profile_backend.did` file in a text editor.
+
+Copy and paste the following `Profile` type declaration and `service` definition for the `getSelf`, `update`, `get`, and `search` functions.
+
+```did
+type Profile = record {
+    "name": text;
+    "description": text;
+    "keywords": vec text;
+};
+
+service : {
+    "getSelf": () -> (Profile) query;
+    "get": (text) -> (Profile) query;
+    "update": (Profile) -> ();
+    "search": (text) -> (opt Profile) query;
+}
+```
+
+Save your changes and close the file to continue.
+
+## Writing the Cargo.toml file
+
+As any standard Rust crate, it has a `Cargo.toml` file which configures the details to build the Rust crate.
+
+Open the `src/rust_profile_backend/Cargo.toml` file and replace the existing content with the following:
+
+``` toml
+[package]
+name = "rust_profile_backend"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+candid = "0.8.2"
+ic-cdk = "0.7.0"
+ic-cdk-macros = "0.6.0"
+serde = { version = "1.0", features = ["derive"] }
+```
+
+Save the file. 
+
+## Update the Cargo.toml dependencies
+
+Since we made changes to the Cargo.toml file, run the following command to update the project's dependencies:
+
+```
+cargo update
+```
+
```

Comment:
```suggestion
```

The Cargo.toml modification is repeated from the 'Creating a new project' section, and the `cargo update` modification isn't needed without it.